### ISSUE TITLE
[Feature] Support `extensions.worktreeconfig` for `repositoryformatversion=1`

### ISSUE THREAD

**Issue by Whirlwind** [libgit2/libgit2#6044]
```
$ git clone --filter=blob:none git@xxx:xx/xx.git
$ cd xx
$ git worktree add ../worktree
$ cd ../worktree
$ git config extensions.worktreeConfig true
```

I have a origin repository with `--filter=blob:none`, and a worktree repository. Then I enable the worktree config in the worktree.

In the libgit2, it could not support the worktree  config with `filter`:

`unsupported extension name extensions.worktreeconfig`

I check the libgit2:
```
static int check_valid_extension(const git_config_entry *entry, void *payload)
{
	GIT_UNUSED(payload);

	if (!strcmp(entry->name, "extensions.noop"))
		return 0;

	git_error_set(GIT_ERROR_REPOSITORY, "unsupported extension name %s", entry->name);
	return -1;
}

static int check_extensions(git_config *config, int version)
{
	if (version < 1)
		return 0;

	return git_config_foreach_match(config, "^extensions\\.", check_valid_extension, NULL);
}
```

And I check the version: `repositoryformatversion = 1`, So all extension configs will failed.

**Comment by vermiculus** [comment]
Is there a to-do list or anything of the sort for this? Or is this as simple as updating `git_config_level_t` to insert some `GIT_CONFIG_LEVEL_WORKTREE` and updating users of that type?

**Comment by bierbaum** [comment]
While testing this, I found the following bug: https://github.com/libgit2/libgit2/pull/6246

After that patch is merged (or integrated here) libgit2 is able to open repos with `extensions.worktreeconfig` set.

**Comment by frankcbliu** [comment]
the same problem, is there any support plan for this?

**Comment by vermiculus** [comment]
@frankcbliu I've got a PR in progress mentioned above, though it's having some as-yet-unknown test failures on some platforms. Any help for this would be welcomed.

**Comment by hcoona** [comment]
> @frankcbliu I've got a PR in progress mentioned above, though it's having some as-yet-unknown test failures on some platforms. Any help for this would be welcomed.

@vermiculus May I know where th PR is? Is it this one? https://github.com/vermiculus/libgit2/tree/sallred/support-worktree-config

**Comment by ethomson** [comment]
Fixed via #6756
